solidity/contracts/periphery/StealthVault.sol

```diff
@@ -0,0 +1,230 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity 0.8.19;
+
+import {ReentrancyGuard} from 'openzeppelin/security/ReentrancyGuard.sol';
+import {IERC20, SafeERC20} from 'openzeppelin/token/ERC20/utils/SafeERC20.sol';
+import {_NATIVE_TOKEN} from '../../utils/Constants.sol';
+
+import {IStealthVault} from '../../interfaces/periphery/IStealthVault.sol';
+
+/*
+ * StealthVault
+ */
+contract StealthVault is IStealthVault, ReentrancyGuard {
+  using SafeERC20 for IERC20;
+
+  address public owner;
+  address public pendingOwner;
+  address public manager;
+  address public pendingManager;
+  uint256 public totalBonded;
+
+  uint256 public gasBuffer = 69_420;
+  uint256 public penalty = 1 ether;
+
+  mapping(address _caller => uint256 _amount) public bonded;
+  mapping(address _caller => uint256 _unboundAt) public canUnbondAt;
+  mapping(bytes32 _hash => address _reportedBy) public hashReportedBy;
+
+  constructor(address _owner, address _manager) {
+    owner = _owner;
+    manager = _manager;
+  }
+
+  function isStealthVault() external pure returns (bool) {
+    return true;
+  }
+
+  function changeOwner(address _pendingOwner) external onlyOwner {
+    pendingOwner = _pendingOwner;
+    emit ChangeOwner(_pendingOwner);
+  }
+
+  function acceptOwner() external onlyPendingOwner {
+    pendingOwner = address(0);
+    owner = msg.sender;
+    emit AcceptOwner(msg.sender);
+  }
+
+  function changeManager(address _pendingManager) external onlyOwner {
+    pendingManager = _pendingManager;
+    emit ChangeManager(_pendingManager);
+  }
+
+  function acceptManager() external onlyPendingManager {
+    manager = pendingManager;
+    pendingManager = address(0);
+    emit AcceptManager(msg.sender);
+  }
+
+  function setPenalty(uint256 _penalty) external onlyOwner {
+    if (_penalty == 0) revert StealthVault_ZeroPenalty();
+    penalty = _penalty;
+    emit PenaltySet(_penalty);
+  }
+
+  function bond() external payable nonReentrant {
+    if (msg.value == 0) revert StealthVault_ZeroAmount();
+    uint256 _bondedAmount = bonded[msg.sender] + msg.value;
+    bonded[msg.sender] = _bondedAmount;
+    totalBonded = totalBonded + msg.value;
+    emit Bonded(msg.sender, msg.value, _bondedAmount);
+  }
+
+  function unbondAll() external {
+    unbond(bonded[msg.sender]);
+  }
+
+  function startUnbond() public nonReentrant {
+    canUnbondAt[msg.sender] = block.timestamp + 4 days;
+  }
+
+  function cancelUnbond() public nonReentrant {
+    canUnbondAt[msg.sender] = 0;
+  }
```

Comment:
there is no incentive for bonding, and not calling `startUnbond`

i'm a keeper, needs to `bond`, i bond my 1 eth and immediately call `startUnbond`, whenever 4 days have passed (or more) i can withdraw my bond instantly

we need to make it so that this cannot happen, create an issue about it an let's address it on a different PR (related to the stealth relayer)